src/courses/views.py

```diff
@@ -65,6 +65,14 @@ def mark_calculations(request, user_id=None):
     # combine assigned_ranges and all_ranges, then order by min mark and only include markranges with unique minimum marks in context queryset
     markranges = (assigned_ranges | all_ranges).order_by('minimum_mark').distinct('minimum_mark')
 
+    # inject the xp needed for passing the mark range
+    # cant do it with template tags as you can only multiply/divide once
+    days_percentage = course_student.semester.fraction_complete()
+    total_xp = courses.first().course.xp_for_100_percent
+    for markrange in markranges:
+        mark_percentage = markrange.minimum_mark / 100
+        markrange.xp_needed = round(total_xp * mark_percentage * days_percentage)
```

Comment:
`Round` is not the same as `Floor` (thank you for providing the reference in your PR notes!).  These will give different results for 34.6  one will be 34 and the other will be 35.  Please use Floor or round down in your python too.

Reply:
done